templates/oath.sh.j2

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+#
+# Sample script to verify MFA using oath-tool
+
+passfile=$1
+
+# Get the user/pass from the tmp file
+user=$(head -1 $passfile)
+pass=$(tail -1 $passfile) 
+
+# Find the entry in our oath.secrets file, ignore case
+secret=$(grep -i -m 1 "$user:" {{ openvpn_mfa_oathsecrets_file }} | cut -d: -f2)
+
+# Calculate the code we should expect
+code=$(oathtool --totp $secret)
+
+if [ "$code" = "$pass" ];
+then
+	exit 0
+fi
```

Comment:
This is safe-ish. Only user gets interpolated and that has some character sanitization:

> To protect against a client passing a maliciously formed username or password string, the username string must consist only of these characters: alphanumeric, underbar ('_'), dash ('-'), dot ('.'), or at ('@'). The password string can consist of any printable characters except for CR or LF. Any illegal characters in either the username or password string will be converted to underbar ('_').

`auth-user-pass-verify` in https://openvpn.net/community-resources/reference-manual-for-openvpn-2-0/

I'd much prefer this to be in a script to keep the values as values instead of potentially executing them.